src/test/java/net/kaczmarzyk/NotInE2eTest.java

```diff
@@ -101,55 +148,99 @@ public void findsByListOfNotAllowedLongValues() throws Exception {
+				.andExpect(jsonPath("$[3].firstName").value("Moe"))
+				.andExpect(jsonPath("$[4].firstName").value("Ned"))
+				.andExpect(jsonPath("$[5]").doesNotExist());
 	}
 
 	@Test
 	public void findsByListOfNotAllowedEnumValues() throws Exception {
 		mockMvc.perform(get("/not-in/customers")
 				.param("genderNotIn", "MALE")
 				.accept(MediaType.APPLICATION_JSON))
-			.andExpect(status().isOk())
-			.andExpect(jsonPath("$").isArray())
-			.andExpect(jsonPath("$.[?(@.firstName=='Marge')]").exists())
-			.andExpect(jsonPath("$.[?(@.firstName=='Lisa')]").exists())
-			.andExpect(jsonPath("$.[?(@.firstName=='Maggie')]").exists())
-			.andExpect(jsonPath("$[4]").doesNotExist());
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$").isArray())
+				.andExpect(jsonPath("$.[?(@.firstName=='Marge')]").exists())
+				.andExpect(jsonPath("$.[?(@.firstName=='Lisa')]").exists())
+				.andExpect(jsonPath("$.[?(@.firstName=='Maggie')]").exists())
+				.andExpect(jsonPath("$[4]").doesNotExist());
+	}
+
+	@Test
+	public void findsByListOfNotAllowedEnumValuesUsingParamSeparator() throws Exception {
+		mockMvc.perform(get("/not-in/customers-ps?genderNotIn=MALE")
+				.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$").isArray())
+				.andExpect(jsonPath("$.[?(@.firstName=='Marge')]").exists())
+				.andExpect(jsonPath("$.[?(@.firstName=='Lisa')]").exists())
+				.andExpect(jsonPath("$.[?(@.firstName=='Maggie')]").exists())
+				.andExpect(jsonPath("$[4]").doesNotExist());
 	}
-	
+
 	@Test
 	public void ignoresUnparseableIntsWhenFilteringOnIntProperty() throws Exception {
-	    mockMvc.perform(get("/not-in//customers")
-                .param("idNotIn", homerSimpson.getId().toString(), "abc")
-                .accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().isOk())
-            .andExpect(jsonPath("$").isArray())
-            .andExpect(jsonPath("$[0].firstName").value("Marge"))
-            .andExpect(jsonPath("$[1].firstName").value("Bart"))
-            .andExpect(jsonPath("$[2].firstName").value("Lisa"))
-            .andExpect(jsonPath("$[3].firstName").value("Maggie"))
-            .andExpect(jsonPath("$[4].firstName").value("Moe"))
-            .andExpect(jsonPath("$[5].firstName").value("Ned"))
-            .andExpect(jsonPath("$[6]").doesNotExist());
+		mockMvc.perform(get("/not-in//customers")
+				.param("idNotIn", homerSimpson.getId().toString(), "abc")
+				.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$").isArray())
+				.andExpect(jsonPath("$[0].firstName").value("Marge"))
+				.andExpect(jsonPath("$[1].firstName").value("Bart"))
+				.andExpect(jsonPath("$[2].firstName").value("Lisa"))
+				.andExpect(jsonPath("$[3].firstName").value("Maggie"))
+				.andExpect(jsonPath("$[4].firstName").value("Moe"))
+				.andExpect(jsonPath("$[5].firstName").value("Ned"))
+				.andExpect(jsonPath("$[6]").doesNotExist());
 	}
 }
```

Comment:
General note: you are missing a test which would check that when paramSeparator is not specified, then values such as "MALE,FEMALE" will not be handled as a multiple values. This should be tested at unit level (SimpleSpecificationResolver) and in one e2e test level (actual controller without paramSeparator specified)

Reply:
When you remove regex support, I would explicitly test it too